File: data/code_pairs/fortran/1481.f90
MODULE valout_mod
  IMPLICIT NONE
CONTAINS
  SUBROUTINE VALOUT(STRING, IBEFOR, WIDTH, VALUE)
    CHARACTER(LEN=*), INTENT(INOUT) :: STRING
    INTEGER, INTENT(INOUT) :: IBEFOR
    INTEGER, INTENT(IN) :: WIDTH
    INTEGER, INTENT(IN) :: VALUE
    CHARACTER(LEN=:), ALLOCATABLE :: formattedStr
    INTEGER :: neededLength

    WRITE(formattedStr, '(I0)') VALUE
    STRING = TRIM(ADJUSTL(formattedStr))
    IBEFOR = LEN_TRIM(formattedStr)
  END SUBROUTINE VALOUT
END MODULE valout_mod

PROGRAM test_valout
  USE valout_mod
  IMPLICIT NONE
  CHARACTER(LEN=100) :: testString
  INTEGER :: testIbefor, width, value

  ! Test 1
  testString = ""
  testIbefor = 0
  width = 5
  value = 123
  CALL VALOUT(testString, testIbefor, width, value)
  PRINT *, "Test 1: ", testString, testIbefor

  ! Test 2
  testString = ""
  testIbefor = 0
  width = 5
  value = -45
  CALL VALOUT(testString, testIbefor, width, value)
  PRINT *, "Test 2: ", testString, testIbefor
END PROGRAM test_valout

File: data/code_pairs/cpp/1481.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <iomanip>
#include <cassert>

class ValoutMod {
public:
    void VALOUT(std::string& STRING, int& IBEFOR, const int WIDTH, const int VALUE) {
        std::ostringstream ss;
        ss << VALUE; // Directly insert VALUE without setting width as it's not utilized
        STRING = ss.str();
        IBEFOR = STRING.length(); // Set IBEFOR to the length of the formatted string
    }
};

void testVALOUT() {
    ValoutMod valout;
    std::string testString;
    int testIbefor, width, value;

    // Test 1
    testString = "";
    testIbefor = 0;
    width = 5; // Note: Width is ignored in this C++ implementation
    value = 123;
    valout.VALOUT(testString, testIbefor, width, value);
    assert(testString == "123" && testIbefor == 3);
    std::cout << "Test 1 passed: " << testString << ", " << testIbefor << std::endl;
    
    // Test 2
    testString = "";
    testIbefor = 0;
    width = 5; // Note: Width is ignored in this C++ implementation
    value = -45;
    valout.VALOUT(testString, testIbefor, width, value);
    assert(testString == "-45" && testIbefor == 3);
    std::cout << "Test 2 passed: " << testString << ", " << testIbefor << std::endl;
}

int main() {
    testVALOUT();
    std::cout << "All tests passed." << std::endl;
    return 0;
}